spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractPollingEndpoint.java

```diff
@@ -266,7 +267,17 @@ private boolean doPoll() {
 			if (holder != null) {
 				holder.setMessage(message);
 			}
-			this.handleMessage(message);
+			try {
+				this.handleMessage(message);
+			}
+			catch (Exception e) {
+				if (e instanceof MessagingException) {
+					throw new OriginalMessageContainingMessagingException(message, (MessagingException) e);
+				}
+				else {
+					throw new MessagingException(message, e);
+				}
```

Comment:
I think that would be difficult for people to understand and it would be a big breaking change.

With this solution, people who don't care about the original message will be oblivious to it.